### DISCUSSION TITLE
display diagnostic count for all open buffers and all buffers with diagnostics in the picker

### DISCUSSION THREAD

**Discussion by bindsdev** [helix-editor/helix#10234] [General]
Buffers, next to the file name, should show the number of errors and/or warning present in the current file. The number could be colored red. If there are more than 9 errors and/or warnings, "9+" could just be displayed, similar to on VSCode.

**Comment by the-mikedavis** [comment]
Is this about the bufferline or the buffer picker?

**Comment by bindsdev** [comment]
> Is this about the bufferline or the buffer picker?

Bufferline. Although it could be another idea to have files with errors or warnings be colored red in the buffer picker.

**Comment by bindsdev** [comment]
> > Is this about the bufferline or the buffer picker?
> 
> Bufferline. Although it could be another idea to have files with errors or warnings be colored red in the buffer picker.

I just realized that the buffer picker existed recently. A similar idea could still be implemented for the buffer picker. Buffers with errors and/or warnings in them can be highlighted red (as per my previous suggestion) in addition with the number of errors/warnings being displayed next to the path to the file. In the picker, since there is significantly more space, we could display the true number of warnings/errors without having to do a "9+".
